dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb

```diff
@@ -12,9 +14,22 @@ def workshop_survey_report
       survey_report[:all_my_workshops_for_course] = get_score_for_workshops(all_my_workshops)
 
       aggregate_for_all_workshops = JSON.parse(AWS::S3.download_from_bucket('pd-workshop-surveys', "aggregate-workshop-scores-#{CDO.rack_env}"))
-      survey_report[:all_workshops_for_course] = aggregate_for_all_workshops[@workshop.course]
+      survey_report[:all_workshops_for_course] = aggregate_for_all_workshops[@workshop.course].symbolize_keys
 
-      render json: survey_report
+      respond_to do |format|
+        format.json do
+          render json: survey_report
+        end
+        format.csv do
+          # Kind of lame but we need to do this - Ruby orders hashes based on insertion order. We want to rename the first
+          # key, but that's not really supported in a way to preserve insertion order. So we have to make a new hash
+          ordered_survey_report = Hash.new
+          ordered_survey_report[@workshop.friendly_name] = survey_report[:this_workshop]
+          survey_report.delete(:this_workshop)
+          ordered_survey_report.merge!(survey_report)
+          send_as_csv_attachment(convert_to_csv(ordered_survey_report), 'workshop_survey_report.csv')
```

Comment:
[Hash.delete](https://docs.ruby-lang.org/en/2.3.0/Hash.html#method-i-delete) returns the deleted value, so lines 27 & 28 could be combined.

Even better, rails provides [Hash.transform_keys](http://api.rubyonrails.org/classes/Hash.html#method-i-transform_keys), so this whole section could be done via:

```ruby
ordered_survey_report = survey_report.transform_keys.with_index{|k, i| i == 0 ? @workshop.friendly_name : k}
```